src/File.php

```diff
@@ -165,14 +164,10 @@ public function addFileHeader(): void
             // Sets Bit 11: Language encoding flag (EFS).  If this bit is set,
             // the filename and comment fields for this file
             // MUST be encoded using UTF-8. (see APPENDIX D)
-            if (!mb_check_encoding($name, 'UTF-8') ||
-                !mb_check_encoding($comment, 'UTF-8')) {
-                throw new EncodingException(
-                    'File name and comment should use UTF-8 ' .
-                    'if one of them does not fit into ASCII range.'
-                );
+            if (mb_check_encoding($name, 'UTF-8') &&
+                mb_check_encoding($comment, 'UTF-8')) {
+                $this->bits |= self::BIT_EFS_UTF8;
```

Comment:
The bit is only set if both name & comment are valid utf8. What happens if they have different encodings?

Reply:
@maennchen 

> The bit is only set if both name & comment are valid utf8.

https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

        Bit 11: Language encoding flag (EFS).  If this bit is set,
                the filename and comment fields for this file
                MUST be encoded using UTF-8. (see APPENDIX D)


> What happens if they have different encodings?

APPENDIX D - Language Encoding (EFS)

...
D.2 If general purpose bit 11 is unset, the file name and comment **SHOULD** conform
to the original ZIP character encoding. ...

Reply:
@mtak3 Should we raise if only one of the texts is not valid utf8?

Reply:
@maennchen 

https://github.com/maennchen/ZipStream-PHP/pull/222/files#diff-987d86d1b2e78bb7723a8e4209c1fb926baa2df9f4a1f33bdee8266814ea573e

ZipArchive allows invalid encoding.

I think so don't have to do anything.